packages/tanstack-react-query/test/useSubscription.test.tsx

```diff
@@ -73,11 +77,18 @@ const getCtx = (protocol: 'http' | 'ws') => {
     onEventIterable: t.procedure
       .input(z.number())
       .subscription(async function* (opts) {
-        for await (const event of on(ee, 'data', {
-          signal: opts.signal,
-        })) {
-          const data = event[0] as number;
-          yield data + opts.input;
+        try {
+          for await (const event of on(ee, 'data', {
+            signal: opts.signal,
+          })) {
+            const data = event[0] as number;
+            yield data + opts.input;
+          }
+        } catch (err) {
+          if (isAbortError(err)) {
+            abortState[opts.input] = 'aborted';
+          }
+          throw err;
         }
```

Comment:
This catch block *shouldn't* be needed

Reply:
It's only there to mark when specific procedure IDs are aborted. onReqAborted is a lot for confusing IMO, but I can switch over to using it if you'd prefer